src/software/new_geom/polynomial2d_test.cpp

```diff
@@ -166,3 +177,16 @@ TEST(Polynomial2dTest, addition_assignment_operator)
 
     EXPECT_EQ(expected, p2);
 }
+
+TEST(Polynomial2dTest, subtraction_assignment_operator)
+{
+    const Polynomial2d p1(Polynomial1d({1, 2, 3}), Polynomial1d({1, 1, 1}));
+
+    Polynomial2d p2(Polynomial1d({6, 8, 10}), Polynomial1d({5, 2, 3}));
+
+    const Polynomial2d expected(Polynomial1d({5, 6, 7}), Polynomial1d({4, 1, 2}));
+
+    p2 -= p1;
+
+    EXPECT_EQ(expected, p2);
+}
```

Comment:
Could we add two tests for the equality operator; equal and not equal?

Reply:
There are three tests  in `polynomial2d_test.cpp`.
`equality_operator_equal` tests the equal.
`equality_operator_x_poly_not_equal` and `equality_operator_y_poly_not_equal` test not equal.

Reply:
My bad, thanks!